nixos/modules/services/x11/xserver.nix

```diff
@@ -329,12 +351,27 @@ in
 
       xrandrHeads = mkOption {
         default = [];
-        example = [ "HDMI-0" "DVI-0" ];
-        type = with types; listOf string;
+        example = [ "HDMI-0" { output = "DVI-0"; primary = true; monitorConfig = ""; } ];
+        type = with types; listOf (either
+          str
+          (submodule {
+            options.output = str;
+            options.primary = bool;
+            options.monitorConfig = lines;
```

Comment:
These are not valid (sub)module options. As your branch is pushable, I'll try to fix that and get it in regardless of whether we have proper module support for this (as noted by https://github.com/NixOS/nixpkgs/pull/13916#issuecomment-198709244) because I currently don't have time to work on addressing #14053.